moviepy/video/tools/cuts.py

```diff
@@ -240,33 +239,27 @@ def select_scenes(self, match_thr, min_time_span, nomatch_thr=None,
                                   if (end-start)>min_time_span]
             
             
-            if (great_long_matches == []):
+            if not great_long_matches:
                 continue # No GIF can be made starting at this time
             
-            poor_matches = set([end for (end,d_min, d_max) in ends_distances
-                            if d_min>nomatch_thr])
-            short_matches = [end for end in ends
-                             if (end-start)<=0.6]
+            poor_matches = {end for (end,d_min, d_max) in ends_distances if d_min > nomatch_thr}
+            short_matches = [end for end in ends if (end-start) <= 0.6]
             
-            if len( poor_matches.intersection(short_matches) ) == 0 :
+            if poor_matches & short_matches:
```

Comment:
This line is wrong. Not only does it do the exact opposite, you can't use the `&` on a list and a set.

`if not poor_matches & set(short_matches):` is correct, but I think that `if not poor_matches.intersection(short_matches):` is the most readable, considering how often the `&` is used as a set intersection operator. Feel free to disagree.